test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -2109,6 +2270,11 @@ func checkForFailedRule(t *testing.T, err error) {
 }
 
 func checkFailureReason(t *testing.T, err error, expectedReason metav1.StatusReason) {
+	if err == nil {
+		// early return, if err is nil but we expect an error, that should be
```

Comment:
Fixed - I had to slightly edit a couple test cases (see: https://github.com/kubernetes/kubernetes/pull/113795/commits/4541ca9cdeff566eb59798b1eac44e10c125da48 ). For those test cases, I think checking for `failureReason` should/did NOP originally before `checkFailureReason` got split off (since there used to be an early exit if the observed and expected errors were identical, `testcase.failureReason` wouldn't get checked), so expected/observed behavior should still be 100% the same.

Reply:
New change where we check err against an empty reason lgtm